providers/harness/pkg/provider.go

```diff
@@ -0,0 +1,310 @@
+				},
+			}
+		} else {
+			return of.StringResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewGeneralResolutionError(generalError),
+					Reason:          of.ErrorReason,
+				},
+			}
+		}
+	}
+
+	harnessTarget, err := toHarnessTarget(evalCtx)
+	if err != nil {
+		return of.StringResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewInvalidContextResolutionError(err.Error()),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+
+	res, _ := p.harnessClient.StringVariation(flag, harnessTarget, defaultValue)
+	return of.StringResolutionDetail{
+		Value:                    res,
+		ProviderResolutionDetail: of.ProviderResolutionDetail{},
+	}
+}
+
+func (p *Provider) ObjectEvaluation(ctx context.Context, flag string, defaultValue interface{}, evalCtx of.FlattenedContext) of.InterfaceResolutionDetail {
+	if p.status != of.ReadyState {
+		if p.status == of.NotReadyState {
+			return of.InterfaceResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewProviderNotReadyResolutionError(providerNotReady),
+					Reason:          of.ErrorReason,
+				},
+			}
+		} else {
+			return of.InterfaceResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewGeneralResolutionError(generalError),
+					Reason:          of.ErrorReason,
+				},
+			}
+		}
+	}
+
+	harnessTarget, err := toHarnessTarget(evalCtx)
+	if err != nil {
+		return of.InterfaceResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewInvalidContextResolutionError(err.Error()),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+
+	defaultValueJson, ok := defaultValue.(types.JSON)
+	if !ok {
+		return of.InterfaceResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewInvalidContextResolutionError("Could not get defaultValue as JSON map"),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+	res, _ := p.harnessClient.JSONVariation(flag, harnessTarget, defaultValueJson)
+	return of.InterfaceResolutionDetail{
+		Value:                    res,
+		ProviderResolutionDetail: of.ProviderResolutionDetail{},
+	}
+}
+
+func toHarnessTarget(evalCtx of.FlattenedContext) (*evaluation.Target, error) {
+	if len(evalCtx) == 0 {
+		return &evaluation.Target{}, nil
+	}
+
+	harnessTarget := &evaluation.Target{}
+
+	custom := make(map[string]interface{})
+	for key, origVal := range evalCtx {
+		val, ok := toStr(origVal)
+		if !ok {
+			return nil, fmt.Errorf("key `%s` can not be converted to string", key)
+		}
+
+		switch key {
+		case "Identifier":
+			harnessTarget.Identifier = val
+		case "Name":
+			harnessTarget.Name = val
+		default:
```

Comment:
if `Name` or `Identifier` are special fields for identifying a target/user, we might want to map the `targetingKey` into one of these, and document which it is. We also may want to provide an option for which it's mapped to, if both are used for that purpose.